src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -301,7 +301,26 @@ private static bool TryProcessTestDataSourceTests(UnitTestElement test, TestMeth
     {
         foreach (var dataSource in testDataSources)
         {
-            var data = dataSource.GetData(methodInfo);
+            IEnumerable<object?[]>? data = null;
+            try
+            {
+                data = dataSource.GetData(methodInfo);
+            }
+            catch (Exception ex)
+            {
+                if (ex is ArgumentException && MSTestSettings.CurrentSettings.MarkTestsWithMissingDynamicDataAsInconclusive)
```

Comment:
Please use a `when` clause in the `catch` instead of this if/else. https://learn.microsoft.com/dotnet/csharp/language-reference/statements/exception-handling-statements#a-when-exception-filter